tests/test_payments.py

```diff
@@ -109,3 +119,86 @@ def test_crud_create_garbage(self, mock_ps_add):
+        new_json = json.loads(resp.data)
+        self.assertEqual (new_json['nickname'], 'favcredit')
+
+# passing correct data to patch
+    @mock.patch.object(PaymentService, 'update_payment', return_value=PATCH_RETURN, autospec=True)
+    def test_crud_update_patch(self,mock_ps_update):
+        print 'here'
+        data = json.dumps(PATCH_CREDIT)
+        resp = self.app.patch('/payments/1', data=data, content_type='application/json')
+        mock_ps_update.assert_called_with(mock.ANY,1,payment_attributes=PATCH_CREDIT)
+        self.assertEqual( resp.status_code, status.HTTP_200_OK )
+        new_json = json.loads(resp.data)
+        self.assertEqual (new_json['nickname'], 'boringcredit')
+
+# passing text data to put
+    def test_crud_update_put_with_text_data(self):
+        resp = self.app.put('/payments/1', data="hello", content_type='text/plain')
+        self.assertEqual( resp.status_code, status.HTTP_400_BAD_REQUEST )
+        self.assertTrue('bad or no data' in resp.data)
+
+# passing text data to patch
+    def test_crud_update_patch_with_text_data(self):
+        resp = self.app.patch('/payments/1', data="hello", content_type='text/plain')
+        self.assertEqual( resp.status_code, status.HTTP_400_BAD_REQUEST )
+        self.assertTrue('bad or no data' in resp.data)
+
+# passing no data to put
+    def test_crud_update_put_with_no_data(self):
+        resp = self.app.put('/payments/1', data=None, content_type='application/json')
+        self.assertEqual( resp.status_code, status.HTTP_400_BAD_REQUEST )
+        self.assertTrue('no data' in resp.data)
+
+# passing no data to patch
+    def test_crud_update_patch_with_no_data(self):
+        resp = self.app.patch('/payments/1', data=None, content_type='application/json')
+        self.assertEqual( resp.status_code, status.HTTP_400_BAD_REQUEST )
+        self.assertTrue('no data' in resp.data)
+
+# passing garbage data to put
+    @mock.patch.object(PaymentService, 'update_payment', side_effect=DataValidationError, autospec=True)
+    def test_crud_update_put_garbage(self, mock_ps_update):
+        garbage = 'a@$*&@#sdassdc3r 3284723X43&^@!#@*#'
+        data = json.dumps(garbage)
+        try:
+            resp = self.app.put('/payments/1', data=data, content_type='application/json')
+        except DataValidationError as e:
+            self.assertTrue('bad or no data' in e.message)
+            mock_ps_update.assert_called_with(mock.ANY, payment_replacement=None)
+            self.assertEqual(resp.status_code, status.HTTP_400_BAD_REQUEST)
+
+# passing garbage data to put
+    @mock.patch.object(PaymentService, 'update_payment', side_effect=DataValidationError, autospec=True)
+    def test_crud_update_patch_garbage(self, mock_ps_update):
+        garbage = 'a@$*&@#sdassdc3r 3284723X43&^@!#@*#'
+        data = json.dumps(garbage)
+        try:
+            resp = self.app.patch('/payments/1', data=data, content_type='application/json')
```

Comment:
these try/except blocks are broken. i had mine like this previously so apologies if that's where you get the idea.

but you'll see that nothing under the except actually runs because that error isn't actually raised by the patch method. it's raised by the mocked update_payment, but then not handled in the caller.

i noticed this in my own code my placing an assert I knew should fail inside that block. when i ran it, everything passed because that test assertion wasn't even made if the exception is caught.

one way to handle it would be creating a context where the assertion is handled in the test:

```
with self.assertRaises(DataValidationError) as context:
    resp = self.app.patch('/payments/1', data=data, content_type='application/json')

self.assertTrue('your custom error message' in context.exception.message)
```

alternatively, you could encapsulate the CRUD update in a try/except and then handled the error accordingly there and assert in the test that however you handled it is done properly.